src/test/java/teammates/test/cases/ui/pagedata/StudentCommentsPageDataTest.java

```diff
@@ -0,0 +1,339 @@
+                String giverName = bundle.getGiverNameForResponse(question, response);
+                String giverTeamName = bundle.getTeamNameForEmail(response.giverEmail);
+                giverName = bundle.appendTeamNameToName(giverName, giverTeamName);
+    
+                String recipientName = bundle.getRecipientNameForResponse(question, response);
+                String recipientTeamName = bundle.getTeamNameForEmail(response.recipientEmail);
+                recipientName = bundle.appendTeamNameToName(recipientName, recipientTeamName);
+                
+                String responseText = response.getResponseDetails().getAnswerHtml(question.getQuestionDetails());
+                
+                ResponseRow responseRow = 
+                        new ResponseRow(giverName, recipientName, responseText, feedbackResponseCommentRows);
+                responseRows.add(responseRow);
+            }
+            int questionNumber = question.questionNumber;
+            String questionText = bundle.getQuestionText(question.getId());
+            String additionalInfo = 
+                    question.getQuestionDetails().getQuestionAdditionalInfoHtml(question.questionNumber, "");
+            QuestionTable questionTable = 
+                    new QuestionTable(questionNumber, questionText, additionalInfo, responseRows);
+            questionTables.add(questionTable);
+        }
+        
+        FeedbackSessionRow sessionRow = 
+                new FeedbackSessionRow(session.feedbackSessionName, session.courseId, questionTables);
+    
+        return sessionRow;
+    }
+
+    private static String getRecipientNames(Set<String> recipients, String studentEmail, CourseRoster roster) {
+        StringBuilder namesStringBuilder = new StringBuilder();
+        int i = 0;
+        
+        for (String recipient : recipients) {
+            if (i == recipients.size() - 1 && recipients.size() > 1) {
+                namesStringBuilder.append(" and ");
+            } else if ( i > 0 && i < recipients.size() - 1 && recipients.size() > 2) {
+                namesStringBuilder.append(", ");
+            }
+            StudentAttributes student = roster.getStudentForEmail(recipient);
+            if (recipient.equals(studentEmail)) {
+                namesStringBuilder.append("you");
+            } else if (data.getCourseId().equals(recipient)) { 
+                namesStringBuilder.append("All Students In This Course");
+            } else if(student != null){
+                if (recipients.size() == 1) {
+                    namesStringBuilder.append(student.name + " (" + student.team + ", " + student.email + ")");
+                } else {
+                    namesStringBuilder.append(student.name);
+                }
+            } else {
+                namesStringBuilder.append(recipient);
+            }
+            i++;
+        }
+        String namesString = namesStringBuilder.toString();
+        return namesString;
+    }
```

Comment:
I checked and found out that this method has appeared in both `InstructorCommentsPageData` and `StudentCommentsPageData`. Can you find a way to extract them out, maybe to `PageData`? It doesn't need to be on this PR, though.